app/models/pro_account.rb

```diff
@@ -29,20 +27,22 @@ def stripe_customer
     @stripe_customer ||= stripe_customer!
   end
 
-  def update_email_address
-    return unless stripe_customer
-    stripe_customer.email = user.email
+  def update_stripe_customer
+    return unless feature_enabled?(:pro_pricing)
+
+    @stripe_customer = stripe_customer || Stripe::Customer.new
```

Comment:
I find this a bit confusing – doesn't seem like we should be building a `Stripe::Customer` here, but can't quite get my head around what problem this prevents.

Reply:
Yeah I see what you mean. The idea here was to create only when object is actually needed. #update_stripe_customer is that point and in the next PR the subscription controller will be using this method too which drastically simplifies our code there.

Reply:
Looking at the diff this is basically what happened in the old `#set_stripe_customer_id` before creation callback

Reply:
Ah just remembered I previously attempted to have this in the `stripe_customer!` method but this broke a seemingly random controller spec in  `alaveteli_pro/subscriptions#destroy`